sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -88,7 +86,14 @@ internal static string GenerateUserAgentString(Assembly clientAssembly, string?
                 version = version.Substring(0, hashSeparator);
             }
             runtimeInformation ??= new RuntimeInformationWrapper();
-            var platformInformation = EscapeProductInformation($"({runtimeInformation.FrameworkDescription}; {runtimeInformation.OSDescription})");
+
+            // RFC 9110 section 5.5 https://www.rfc-editor.org/rfc/rfc9110.txt#section-5.5 does not require any specific encoding : "Fields needing a greater range of characters
+            // can use an encoding, such as the one defined in RFC8187." RFC8187 is targeted at parameter values, almost always filename, so using url encoding here instead, which is
+            // more widely used. Since user-agent does not usually contain non-ascii, only encode when necessary.
+            // This was added to support operating systems with non-ascii characters in their release names.
+            var osDescription = ContainsNonAscii(runtimeInformation.OSDescription) ? WebUtility.UrlEncode(runtimeInformation.OSDescription) : runtimeInformation.OSDescription;
```

Comment:
Also curious -- if this only happens on a specific build of Linux, would it make sense to scope the check to that?  Or do we feel it's preferable to be fully general to be robust to the future?

Reply:
I think it is preferable to be more robust, this is the only one I know of but the change in the RuntimeInformation type to prefer the pretty name was fairly recent, so I wonder if this may be something that could come up occasionally